cmd/dep/status.go

```diff
@@ -1068,6 +1068,33 @@ func collectConstraints(ctx *dep.Ctx, p *dep.Project, sm gps.SourceManager) (con
 		}
 	}
 
+	// Incorporate constraints set in the manifest of the root project.
+	if p.Manifest != nil {
+
+		ineffCon := p.FindIneffectualConstraints(sm)
```

Comment:
We don't need this. We can use `directDeps` that we have above in this function. This is because, any constraint in manifest that's a direct dependency is strictly applied. If we can't apply it, dep's solver would fail.